### PR TITLE
feat(nav): offer dark theme version

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-next/issues/2192

### CODE REVIEW

**Comment by christiemolloy** [comment]
@mcoker seeing this:

Screen Shot 2019-09-04 at 5 12 29 PM

**Comment by mcoker** [comment]
@kybaker @lboehling @mcarrano I've updated all of the examples and demos, should be good for you to review now.

File: src/patternfly/components/Nav/docs/code.md [hunk 1]
```diff
@@ -24,23 +24,24 @@ The navigation system relies on several different sub-components:
...
+| `.pf-c-nav__toggle` | `<span>` | Initiates a chevron indicating expandability of a `pf-c-nav__list-link`. |
 | `.pf-c-nav__toggle-icon` | `<i>` | Initiates a nav toggle icon. |
-| `.pf-m-expandable` | `.pf-c-nav__item` | Modifies for the expandable state |
-| `.pf-m-expanded` | `.pf-c-nav__item` | Modifies for the expanded state |
-| `.pf-m-hover` | `.pf-c-nav__link` | Modifies to display the link as hovered |
-| `.pf-m-focus` | `.pf-c-nav__link` | Modifies to display the link as focussed |
-| `.pf-m-current` | `.pf-c-nav__link` | Modifies for the current state |
-| `.pf-m-active` | `.pf-c-nav__link` | Modifies to display the link as active |
-| `.pf-m-disabled` | `.pf-c-nav__link` | Modifies to display the link as disabled |
+| `.pf-m-dark` | `.pf-c-nav` | Modifies the nav for the dark variation. **Note: only for use with vertical navs, and requires `.pf-m-dark` on the page component's sidebar element** |
```
Comment (christiemolloy) [inline src/patternfly/components/Nav/docs/code.md]: can you add the class name to the end of this sentence just to be very clear: `pf-c-page__sidebar`

File: src/patternfly/components/Nav/nav.scss [hunk 2]
```diff
@@ -3,6 +3,29 @@
...
+  --pf-c-nav--m-dark__simple-list-link--hover--Color: var(--pf-global--Color--light-100);
+  --pf-c-nav--m-dark__simple-list-link--focus--Color: var(--pf-global--Color--light-100);
+  --pf-c-nav--m-dark__simple-list-link--active--Color: var(--pf-global--Color--light-100);
+  --pf-c-nav--m-dark__simple-list-link--hover--BackgroundColor: var(--pf-global--BackgroundColor--dark-400);
+  --pf-c-nav--m-dark__simple-list-link--focus--BackgroundColor: var(--pf-global--BackgroundColor--dark-400);
+  --pf-c-nav--m-dark__simple-list-link--active--BackgroundColor: var(--pf-global--BackgroundColor--dark-400);
+  --pf-c-nav--m-dark__simple-list-link--m-current--Color: var(--pf-global--Color--light-100);
+  --pf-c-nav--m-dark__simple-list-link--m-current--FontWeight: var(--pf-global--FontWeight--normal);
+  --pf-c-nav--m-dark__simple-list-link--m-current--BackgroundColor: var(--pf-global--BackgroundColor--dark-400);
+  --pf-c-nav--m-dark__item--m-expanded__simple-list-link--Color: var(--pf-global--Color--light-100);
```
Comment (christiemolloy) [inline src/patternfly/components/Nav/nav.scss]: Should this be `--pf-c-nav--m-dark__item--m-current-list-link--Color` instead? same for the var below?

File: src/patternfly/components/Nav/docs/code.md [hunk 3]
```diff
@@ -24,23 +24,24 @@ The navigation system relies on several different sub-components:
...
+| `.pf-c-nav__toggle` | `<span>` | Initiates a chevron indicating expandability of a `pf-c-nav__list-link`. |
 | `.pf-c-nav__toggle-icon` | `<i>` | Initiates a nav toggle icon. |
-| `.pf-m-expandable` | `.pf-c-nav__item` | Modifies for the expandable state |
-| `.pf-m-expanded` | `.pf-c-nav__item` | Modifies for the expanded state |
-| `.pf-m-hover` | `.pf-c-nav__link` | Modifies to display the link as hovered |
-| `.pf-m-focus` | `.pf-c-nav__link` | Modifies to display the link as focussed |
-| `.pf-m-current` | `.pf-c-nav__link` | Modifies for the current state |
-| `.pf-m-active` | `.pf-c-nav__link` | Modifies to display the link as active |
-| `.pf-m-disabled` | `.pf-c-nav__link` | Modifies to display the link as disabled |
+| `.pf-m-dark` | `.pf-c-nav` | Modifies the nav for the dark variation. **Note: only for use with vertical navs, and requires `.pf-m-dark` on the page component's sidebar element (`.pf-c-page__sidebar`)** |
```
Comment (mattnolting) [inline src/patternfly/components/Nav/docs/code.md]:
```suggestion
| `.pf-m-dark` | `.pf-c-nav` | Modifies the nav for the dark variation. **Note: only for use with vertical navs, and requires `.pf-m-dark` on the page component's sidebar element (`.pf-c-page__sidebar`)**. |
```

**Pull request merged**